src/agent/agent.c

```diff
@@ -30,6 +31,69 @@ static bool
                                 void *userdata,
                                 free_func_t free_userdata);
 
+#define AGENT_HEARTBEAT_INTERVAL_USEC (1000000)
+
+static int agent_reset_heartbeat_timer(Agent *agent, sd_event_source **event_source);
+
+static int agent_heartbeat_timer_callback(sd_event_source *event_source, UNUSED uint64_t usec, void *userdata) {
+        Agent *agent = userdata;
+        int r = 0;
+
+        assert(event_source);
+
+        // static uint32_t count;
+        // printf("%s(): heartbeat count = %d\n", __func__, count++);
+
+        r = sd_bus_emit_signal(
+                        agent->peer_dbus,
+                        INTERNAL_AGENT_OBJECT_PATH,
+                        INTERNAL_AGENT_INTERFACE,
+                        "Heartbeat",
+                        "s",
+                        agent->name);
+        if (r < 0) {
+                fprintf(stderr, "Failed to emit heartbeat signal: %m");
+                return r;
+        }
+
+        r = agent_reset_heartbeat_timer(agent, &event_source);
+        if (r < 0) {
+                fprintf(stdout, "Failed to reset periodic timer event source, ignoring: %m");
+                return r;
+        }
+
+        return 0;
+}
+
+static int agent_reset_heartbeat_timer(Agent *agent, sd_event_source **event_source) {
+        sd_event *event = agent->event;
```

Comment:
Since you're using `event` only once to pass it to the next function, there's no need to keep a local variable for it

Reply:
Agreed.